tests/s25Main/mapGenerator/testHeadQuarters.cpp

```diff
@@ -165,7 +170,45 @@ BOOST_AUTO_TEST_CASE(PlaceHeadQuarter_places_hq_on_map_at_suitable_position)
     });
 }
 
-BOOST_AUTO_TEST_CASE(PlaceHeadQuarters_returns_true_for_any_player_number_on_suitable_map)
+BOOST_AUTO_TEST_CASE(PlaceHeadQuarter_with_empty_area_throws_exception)
+{
+    MapExtent size(8, 8);
+    RunTest(size, [&size](Map& map, TextureMap& textures) {
+        auto water = textures.Find(IsShipableWater);
+        map.textures.Resize(size, TexturePair(water));
+        std::vector<MapPoint> area;
+        bool exceptionThrown = false;
+        try
+        {
+            PlaceHeadQuarter(map, 0, area);
+        } catch(...)
+        {
+            exceptionThrown = true;
+        }
+        BOOST_REQUIRE(exceptionThrown);
```

Comment:
There is a Boost.Test facility for this: BOOST_REQUIRE_THROW: https://www.boost.org/doc/libs/1_64_0/libs/test/doc/html/boost_test/testing_tools/exception_correctness.html